include/zephyr/bluetooth/audio/bap.h

```diff
@@ -793,6 +793,59 @@ struct bt_bap_scan_delegator_cb {
 	void (*scanning_state)(struct bt_conn *conn, bool is_scanning);
 };
 
+/**
+ * @brief Struct to hold callbacks for the Scan Delegator
+ *
+ */
+struct bt_bap_scan_delegator_cp_cb {
+	/**
+	 * @brief Add Source operation callback
+	 *
+	 * These callbacks notify the application when a request comes
+	 * in to add a source. The application can return BT_ATT_ERR_SUCCESS to
+	 * accept or a valid GATT/ATT error code to reject the request.
+	 *
+	 * @param conn       Pointer to the connection that initiated the request,
+	 *                   or NULL if locally triggered.
+	 * @param recv_state Pointer to the requested receive state to be added.
+	 *
+	 * @return BT_ATT_ERR_SUCCESS to accept, or a valid GATT/ATT error code to reject.
```

Comment:
It needs to be either a GATT or an ATT error code. 

ATT error codes are positive, but `BT_GATT_ERR` negates it. 
For `BT_ATT_ERR_SUCCESS` it doesn't matter, but it does when we return the value in the function. 

Looking at the code, it seems that it shall be a BT_ATT_ERR value, and not a negative value since you always apply `BT_GATT_ERR` on it